app/packs/src/apps/mydb/elements/details/reactions/variationsTab/ReactionVariationsCellComponents.js

```diff
@@ -0,0 +1,139 @@
+/* eslint-disable react/display-name */
+import React from 'react';
+import { AgGridReact } from 'ag-grid-react';
+import {
+  Button, ButtonGroup, Badge
+} from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import {
+  getVariationsRowName,
+  convertUnit
+} from 'src/apps/mydb/elements/details/reactions/variationsTab/ReactionVariationsUtils';
+import {
+  updateNonReferenceMaterialOnMassChange,
+  getReferenceMaterial, getMolFromGram, getGramFromMol
+} from 'src/apps/mydb/elements/details/reactions/variationsTab/ReactionVariationsMaterials';
+
+function RowToolsCellRenderer({
+  data: variationsRow, context
+}) {
+  const { reactionShortLabel, copyRow, removeRow } = context;
+  return (
+    <div>
+      <Badge>{getVariationsRowName(reactionShortLabel, variationsRow.id)}</Badge>
+      {' '}
+      <ButtonGroup>
+        <Button bsSize="xsmall" bsStyle="success" onClick={() => copyRow(variationsRow)}>
+          <i className="fa fa-clone" />
+        </Button>
+        <Button bsSize="xsmall" bsStyle="danger" onClick={() => removeRow(variationsRow)}>
+          <i className="fa fa-trash-o" />
+        </Button>
+      </ButtonGroup>
+    </div>
+  );
+}
+
+RowToolsCellRenderer.propTypes = {
```

Comment:
We thought about extracting `RowToolsCellRenderer` from `ReactionVariationsCellComponents.js`, since it's the only React component in the file. The file otherwise contains only formatters and parsers for the `ag-grid` framework. Extracting the component would mean adding another file exclusively for `RowToolsCellRenderer`. I'd argue that this adds more complexity compared to leaving `RowToolsCellRenderer` in `ReactionVariationsCellComponents.js`, since I believe that, at least conceptually, `RowToolsCellRenderer` fits well into `ReactionVariationsCellComponents.js`.